src/util/effect.ts

```diff
@@ -0,0 +1,21 @@
+import type { Task } from 'fp-ts/Task'
+import type { IO } from 'fp-ts/IO'
+
+// Defines the type of function that returns an IO or a Task also known as a computational function.
+export type ComputationalFunc = (...args: unknown[]) => IO<unknown> | Task<unknown>
+
+// toEffectful is an utility function that converts a computational function (i.e a function that returns an IO or a Task) to a function that returns
+// its result directly (producing a side effect).
+export const toEffectful =
+  <F extends ComputationalFunc> (
+    fn: F
+  ): ((
+    ...args: Parameters<F>
+  ) => ReturnType<F> extends IO<infer T>
+    ? T
+    : ReturnType<F> extends Task<infer U>
+    ? Promise<U>
+    : never) =>
+  (...args: Parameters<F>) =>
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fn(...args)() as any
```

Comment:
I will have used `never` as return type because it represents the unreachable result of the first execution. 
By the way, it's more precise than `any` keyword.

Reply:
See 70ece341b0f447cf42b328c8e33fe45c7217b5fa. And tell me.

Reply:
@ccamel 
Love it!
That's what I was meaning by the never usage: constraint type to narrowing conditionals until the never type.
Thanks